pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/readers/json/ImmutableJsonIndexReader.java

```diff
@@ -76,6 +77,63 @@ public ImmutableJsonIndexReader(PinotDataBuffer dataBuffer, int numDocs) {
     _docIdMapping = dataBuffer.view(invertedIndexEndOffset, docIdMappingEndOffset, ByteOrder.LITTLE_ENDIAN);
   }
 
+  /**
+   * Accepts a JSON key and array of docIds used to filter the response
+   * return a String[] where String[i] gives the value of $.key for document i
+   */
+  @Override
+  public String[] getValuesForKeyAndDocs(String key, int[] docIds) {
+    ImmutableRoaringBitmap docIdMask = ImmutableRoaringBitmap.bitmapOf(docIds);
```

Comment:
From flame graphs, this generally accounts for 30-50% of total time. I didn't find a simple way to optimize this, input is appreciated